frontend/packages/ux-editor/src/components/toolbar/ToolbarItemComponent.tsx

```diff
@@ -24,6 +25,10 @@ export const ToolbarItemComponent = ({
     setCompInfoPanelOpen((prevState) => !prevState);
   };
 
+  if (componentType == 'Summary2' && shouldDisplayFeature('summary2') === false) {
+    return null;
+  }
```

Comment:
This is an okey was to solve this in my opinion, but maybe you and I should align methods? 😅 
I added the `shouldDisplayFeature('FEATURE_FLAG_KEY')` in the list returning the components when adding support for `SubForm` component.

https://github.com/Altinn/altinn-studio/pull/13564/files#diff-7d39354e65b8539e08b00a4a00c1e30b7cad6696eaa9c76a1080c2acc696f433R471

But either way is good I suppose. One of us are going to get a merge conflict either way 😛